helix-term/src/ui/editor.rs

```diff
@@ -548,6 +552,35 @@ impl EditorView {
         );
     }
 
+    /// Apply the highlighting on the lines where a cursor is active
+    pub fn highlight_cursorline(doc: &Document, view: &View, surface: &mut Surface, theme: &Theme) {
+        let text = doc.text().slice(..);
+        let last_line = view.last_line(doc);
+
+        // it's used inside an iterator so the collect isn't needless:
+        // https://github.com/rust-lang/rust-clippy/issues/6164
+        #[allow(clippy::needless_collect)]
+        let cursors: Vec<_> = doc
+            .selection(view.id)
+            .iter()
+            .map(|range| range.cursor_line(text))
+            .collect();
+
+        for line in view.offset.row..(last_line + 1) {
+            if cursors.contains(&line) {
+                surface.set_style(
+                    Rect::new(
+                        view.area.x,
+                        view.area.y + (line - view.offset.row) as u16,
+                        view.area.width,
+                        1,
+                    ),
+                    theme.get("ui.highlight"),
```

Comment:
I think it should also highlight the primary cursor line on a different (visible) windows for the same document, doom emacs did that and I do find it useful. So the behavior is more or less

1. take the current document primary cursor line
2. for all the visible window which is the same document, highlight the same primary cursor line for the window user is on (see 1)
3. for all the other visible window which is a different document, highlight the row for the primary cursor line for that window

Regarding secondary cursors, I don't have much opinion on it, but I think if we want to display it, it should be using a secondary color.

Reply:
This indicates to me that there are 4 possible combinations for what cursorlines to show / how to style cursorlines, based on the following:
- whether it's a primary or secondary cursor
- whether it's from the current window, or another window

Anecdotally, I would only want to show cursorlines from the current window's cursors, but for both primary and secondary cursors, with the primary cursor having a more visible appearance.

Reply:
@pickfire Currently, cursors/selections aren't synced (are they in doom emacs?).
I see 4 ways to handle this:
1. (your suggestion) Don't synchronize cursors/selections, but highlight the lines according to the active window. Might be confusing when a cursorline is visible on the secondary window, but you end up being on another line when actually switching to it.
2. (my suggestion, current implementation) Each window has its own cursorlines. Not synchronized.
3. Don't show cursorlines on inactive windows.
4. Synchronize cursors/selections across windows on the same buffer.

Option 4 is not trivial at all. If desired, it should probably be done in a separate PR.

Reply:
Doom emacs are cursors are synced I believe, since if you move on one window other windows cursor moved as well. To implement that I think we can just put cursorline to document rather than window.